messaging/broker.go

```diff
@@ -1017,6 +1009,19 @@ func NewTopicReader(path string, index uint64, streaming bool) *TopicReader {
 	}
 }
 
+// Seek seeks to a position the current segment.
+func (r *TopicReader) Seek(offset int64, whence int) (int64, error) {
+	assert(whence == os.SEEK_CUR, "topic reader can only seek to a relative position")
```

Comment:
Seems like you could remove this assert by simply having this function ignore `whence`, and then calling `Seek()` with `os.SEEK_CUR`. Of course, you may wish to catch errors in the future.